### ISSUE TITLE
Cursor based pagination

### ISSUE THREAD

**Issue by cloverich** [cloverich/chronicles#34]
When navigating documents, there is no pagination as of now. In a prior iteration, I used a [windowing library](https://github.com/bvaughn/react-window) and date jumping to enable usable infinite scroll. I deferred that feature and until I can properly invest in it believe adding search and basic pagination will be sufficient for base use cases. Ideally this feature would be done after #39 and would include some refactoring around how documents are returned in the API calls.

EDIT: Leaving original context. I think the MVP way to do this is to use before and after tags, and call this "cursor based" pagination. The ultimate goal that I've always wanted is a "window" based pagination. Its like infinite scroll, but its goal is to make it easy to scroll around quickly for searching, similar to Google Photo's.
